components/datetime/Cargo.toml

```diff
@@ -56,8 +56,7 @@ icu = { path = "../icu" }
 icu_benchmark_macros = { version = "0.7", path = "../../tools/benchmark/macros" }
 icu_provider = { version = "1.0.0-beta1", path = "../../provider/core" }
 icu_provider_adapters = { path = "../../provider/adapters" }
-icu_testdata = { path = "../../provider/testdata", default-features = false, features = ["buffer", "icu_datetime", "icu_timezone", "icu_calendar", "icu_decimal", "icu_plurals"] }
-icu_calendar = { version = "1.0.0-beta1", path = "../calendar", features = ["serde"] }
+icu_testdata = { path = "../../provider/testdata", default-features = false, features = ["buffer", "icu_datetime", "icu_datetime_experimental", "icu_timezone", "icu_calendar", "icu_decimal", "icu_plurals"] }
```

Comment:
And what's the problem with that?

Reply:
I want to be able to run tests with the experimental features disabled, as we've discussed previously.

Reply:
But isn't this achieved now by using a different code path if experimental is enabled?

Reply:
I split out `try_new_experimental_unstable` because I thought it was a cleaner solution, but I still think it's very bad to force-enable the experimental feature in a hidden way when running any datetime tests and benches. There is no other feature in any component to my knowledge that we force-enable in dev mode.

Reply:
If you want to cover `try_new_experimental_unstable(icu_provider::unstable)`, you can do that in some other crate like `icu` in a feature-gated integration test.

Reply:
You want to be sure to run the non-experimental code path. But our CI still tests with all features, so that's not going to happen even if testdata doesn't enable it. We should not be trying to turn off the feature, we should just run both code paths when the feature is turned on.

Reply:
I would like to run `cargo test-all-features` on CI but it is too slow, but it's something I think we should try to make work. This PR gets us a lot closer to making it work, which is great. But it goes against the whole spirit of test-all-features if we force-enable the datetime experimental feature when running tests in that crate.

Reply:
Also, although I disentangled the experimental features in the core library code, they are still tangled in the test code. There are code paths in the tests that won't run if the experimental feature is enabled. Sure, I could put in more work to disentangle those, too, but the point is that I _shouldn't have to do that_.

Reply:
> I shouldn't have to do that.

Debatable. test-all-features is very expensive, *I believe* it's better to assume that tests are always running with `--all-features` and cover all code paths. If that's not possible, the features aren't additive and should be fixed.

Reply:
The reason why I don't want to disable the feature is that all the doctests for experimental would need to use `icu_testdata::buffer().as_deserializing()` and that's not something I want to put in documentation.

Reply:
I'm not too concerned about the appearance of experimental docs because they are, well, experimental. But, if you don't want docs tests to show

```
use icu_provider::AsDeserializingBufferProvider;
try_new_experimental_unstable(&icu_testdata::buffer().as_deserializing(), ...
```

then you could apply the macro to the experimental constructors and write

```
try_new_experimental_with_buffer_provider(&icu_testdata::buffer(), ...
```